CustomColors.cpp

```diff
@@ -318,3 +322,170 @@ HOOK_METHOD_PRIORITY(FTLButton, OnRender, 5000, () -> void)
+    if (currentColor.r == 1.f && currentColor.g == 1.f && currentColor.b == 1.f && currentColor.a == 1.f)
+    {
+        // normal; white
+        ret.r = choiceColor->normal.r;
+        ret.g = choiceColor->normal.g;
+        ret.b = choiceColor->normal.b;
+        ret.a = choiceColor->normal.a;
+    }
+    else if (currentColor.r == 0.9529412f && currentColor.g == 1.f && currentColor.b == 0.3137255f && currentColor.a == 1.f)
+    {
+        // hover; yellow
+        ret.r = choiceColor->hover.r;
+        ret.g = choiceColor->hover.g;
+        ret.b = choiceColor->hover.b;
+        ret.a = choiceColor->hover.a;
+    }
+    else if (currentColor.r == 0.5882353f && currentColor.g == 0.5882353f && currentColor.b == 0.5882353f && currentColor.a == 1.f)
+    {
+        // disabled; gray
+        ret.r = choiceColor->disabled.r;
+        ret.g = choiceColor->disabled.g;
+        ret.b = choiceColor->disabled.b;
+        ret.a = choiceColor->disabled.a;
+    }
+    else if (currentColor.r == 0.f && currentColor.g == 0.7647059f && currentColor.b == 1.f && currentColor.a == 1.f)
+    {
+        // blue_option; cyan
+        ret.r = choiceColor->blue_option.r;
+        ret.g = choiceColor->blue_option.g;
+        ret.b = choiceColor->blue_option.b;
+        ret.a = choiceColor->blue_option.a;
+    }
+    return ret;
+}
+
+HOOK_METHOD(EventsParser, ProcessChoice, (EventTemplate *event, rapidxml::xml_node<char> *node, const std::string &eventName) -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> EventsParser::ProcessChoice -> Begin (CustomColors.cpp)\n")
+    if (!CustomOptionsManager::GetInstance()->enableCustomChoiceColors.currentValue || node->first_node("text")->first_attribute("color") == nullptr)
+    {
+        super(event, node, eventName);
+        return;
+    }
+
+    if (node->first_node("text")->first_attribute("id") != nullptr)
+    {
+        node->first_node("text")->first_attribute("id")->value((std::string(node->first_node("text")->first_attribute("id")->value()) + EncodeChoicecColorName(node->first_node("text")->first_attribute("color")->value())).c_str());
+    }
+    else
+    {
+        node->first_node("text")->value((std::string(node->first_node("text")->value()) + EncodeChoicecColorName(node->first_node("text")->first_attribute("color")->value())).c_str());
+    }
+    super(event, node, eventName);
+}
+
+
+HOOK_STATIC(freetype, easy_printAutoNewlines, (int fontSize, float x, float y, int line_length, const std::string &text) -> Pointf)
+{
+    LOG_HOOK("HOOK_STATIC -> freetype::easy_printAutoNewlines -> Begin (CustomColors.cpp)\n")
+    if (!CustomOptionsManager::GetInstance()->enableCustomChoiceColors.currentValue) return super(fontSize, x, y, line_length, text);
+
+    std::regex re("^(.*)\\[\\[#C\\:(.*?)\\]\\]$");
+    std::smatch match;
+    std::string new_text = text;
+    GL_Color currentColor = CSurface::GL_GetColor();
+    if (std::regex_match(new_text, match, re))
```

Comment:
I'm not a huge fan of all these regex checks being made every frame in all of these text printing functions, it seems pretty expensive. Would it be possible to just rewrite the function that renders event choices instead?

Reply:
New check makes it much better, but could be improved in the future.